Topics.md

```diff
@@ -0,0 +1,20 @@
+# Built-in Quality Practices
+
+## Test-Driven Development (TDD) 
+
+TDD is based on three rules, knows as 'The Three Laws of TDD': 
+
+* You must write a failing test before you write any production code.
+* You must not write more of a test than is sufficient to fail, or fail to compile.
+* You must not write more production code than is sufficient to make the currently 
+  failing test pass.
+  
+In code, TDD is translated into the cycle Red/Green/Refactor!
+[![TDD_Cycle](photos/red_green_refactor.jpg)](photos/red_green_refactor.jpg)
+
+The rules of this cycle are as follows: 
+* Create a unit tests that fails
+* Write production code that makes that test pass. 
+* Clean up the mess you just made.
+
+
```

Comment:
What are the top 3 advantages of TDD regarding built-in quality according to you?